File: pinyougouparent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/service/BrandService.java
package com.pinyougou.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * 商品接口
 */
public interface BrandService {

  public List<TbBrand> findAll();

    // 分页查询
   public PageResult findAllPage(Integer pageNum , Integer pageSize);

    /**
     * 添加
     * @param brand
     */
    public   void   insert(TbBrand brand);

    //根据id查询
    public  TbBrand findOne(Long id);

    //修改
    public  void  updata(TbBrand tbBrand);


    //删除

    public  void  delete (Long [] id);


      // 根据条件查询
     public PageResult serach (TbBrand tbBrand, Integer pageNum, Integer pageSize);

     //品牌下拉选框
     List<Map> selectOptionList();


}

File: pinyougouparent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service('brandService', function ($http) {
    // 查询
    this.findall = function () {
        return $http.get("../Brand/findAll.do")
    }
    //分页查询
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../Brand/findAllPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize + ' ')
    }
    // 添加
    this.add = function (brand) {
        return $http.post('../Brand/add.do', brand)
    }
    //修改
    this.updata = function (brand) {
        return $http.post('../Brand/updata.do', brand)
    }
    //根据id查询
    this.findOne = function (id) {
        return    $http.post("../Brand/findOne.do?id=" + id)
    }
    //批量删除
    this.delete = function (ids) {
        return   $http.get('../Brand/delete.do?ids=' + ids)
    }
    //模糊查询
    this.search = function (pageNum,pageSize,serachBrand) {
        return  $http.post('../Brand/serach.do?pageNum=' + pageNum + '&pageSize=' + pageSize, serachBrand)
    }
     //品牌下拉列表
    this.selectOptionList  =function () {
         return $http.get('../Brand/selectOptionList.do');
    }
});